script.js

```diff
@@ -1,108 +1,116 @@
+
+function createCard(episode) {
+  let cardDiv = document.createElement("div");
+  cardDiv.classList.add("cardDiv");
+  let cardHeader = document.createElement("div");
+  cardHeader.classList.add("cardHeader");
+  let episodeTitle = document.createElement("h2");
+  let episodeSeason = document.createElement("h4");
+  let cardImg = document.createElement("img");
+  let summary = document.createElement("p");
+  episodeTitle.textContent = episode.name;
+  episodeSeason.textContent = `${formatedEpisodeNum(
+    episode.season,
+    episode.number
+  )}`;
+  cardImg.src = episode.image.medium;
+  summary.innerHTML = episode.summary;
+  cardHeader.appendChild(episodeTitle);
+  cardHeader.appendChild(episodeSeason);
+  cardDiv.appendChild(cardHeader);
+  cardDiv.appendChild(cardImg);
+  cardDiv.appendChild(summary);
+  rootElem.appendChild(cardDiv);
+}
+
+// Format season and episode numbers to 2 digits //
+
+function formatedEpisodeNum(season, episode) {
+  if (season.toString().length < 2) {
+    season = "0" + season;
+  }
+  if (episode.toString().length < 2) {
+    episode = "0" + episode;
+  }
+  return `S${season}E${episode}`;
+}
 
-  // ADDS SEARCH FEATURE
-  // stugglinh with this function. It was working fine at some point.
-  // Then I dont know waht happened and the letters would take ages to appear when I type
-  // I keep changing the code. When I fix one part the other
-  //  part does not work and vice-versa.
-
-  searchBox.addEventListener("keyup", function (e) {
-    const searchTerm = e.target.value.toLowerCase();
-
-    let episodes = document.querySelectorAll(".card");
-    newArrayOfEpisodes = Array.from(episodes);
-
-    newArrayOfEpisodes.forEach(function (episode) {
-      if (episode.innerText.toLowerCase().includes(searchTerm)) {
-        episode.style.display = "block";
-      } else {
-        episode.style.display = "none";
-      }
-      let filteredEpisodes = newArrayOfEpisodes.filter(function (element) {
-        element.style.display === "block";
-      });
-
-      searchResults.innerText = `Displaying ${filteredEpisodes.length}/${episodes.length} episodes`;
-    });
+// Add search functionality //
+
+searchInput.addEventListener("keyup", searchTerm);
+
+function searchTerm() {
+  let termSearched = searchInput.value.toLowerCase();
+  let matchingEpisodes = allEpisodes.filter((episode) => {
+    return (episode.summary + episode.name)
+      .toLowerCase()
+      .includes(termSearched);
   });
+  makePageForEpisodes(matchingEpisodes);
+}
 
-  // ADDS AN EPISODE SELECTOR
+// Add dropdown menu //
 
-  let selectInput = document.getElementById("selectedEpisode");
+let selectDropdown = document.getElementById("selectedEpisode");
 
+function makeDropdownMenu() {
   for (let i = 0; i < allEpisodes.length; i++) {
     let dropDownList = document.createElement("option");
 
-    selectInput.appendChild(dropDownList);
+    selectDropdown.appendChild(dropDownList);
     Option.value = `S${allEpisodes[i].season
```

Comment:
great, you solve this problem in two ways and both are correct padstart and before that you use function formatedEpisodeNum and both definitely are right